### ISSUE TITLE
New lint: use `slice.first()` instead of `slice.get(0)`

### ISSUE THREAD

**Issue by equal-l2** [rust-lang/rust-clippy#8851]
### What it does

Detects `slice.get(0)` and suggests replacement with `slice.first()`.

### Lint Name

slice_get_zero

### Category

style

### Advantage

No magic numbers.

### Drawbacks

`first()` is longer than `get(0)` by 1 character.

### Example

```rust
let v = vec![0, 1, 2];
println!("{}", vec.get(0));
```

Could be written as:

```rust
let v = vec![0, 1, 2];
println!("{}", vec.first());
```

**Comment by smmoosavi** [comment]
this rule is annoying. In this code, we need the first 3 items in the list.

```rust
let path = list.get(0);
let next = list.get(1);
let other = list.get(2);
```
but `clippy::get_first` is forcing us to change the into:

```rust
let path = list.first();
let next = list.get(1);
let other = list.get(2);
```

which is less readable

**Comment by y21** [comment]
@smmoosavi You can also write that as `let [path, next, other, ..] = &*list else { /* handle the failure case */ };` which is arguably better than both, or you can just `#[allow]` the lint for that let statement if you don't agree or it's not applicable.

In general, I don't think there is anything that can be done about aesthetic cases like yours. That argument can be made about pretty much almost every `clippy::style` lint that exists. E.g.:
```rs
let x = (collection.len() == 0).then_some(foo);
let y = (collection.len() == 1).then_some(bar);
let z = (collection.len() == 2).then_some(baz);
```
Clippy also suggests using is_empty on only the first one, which breaks symmetry and doesn't look as nice 🤷‍♂️
